google/container.go

```diff
@@ -114,17 +114,25 @@ func (c *Container) Put(name string, r io.Reader, size int64, metadata map[strin
 	}
 
 	w := obj.NewWriter(c.ctx)
+	w.ObjectAttrs.Metadata = merge(w.ObjectAttrs.Metadata, mdPrepped)
 	if _, err := io.Copy(w, r); err != nil {
 		return nil, err
 	}
-	w.Close()
-
-	attr, err := obj.Update(c.ctx, storage.ObjectAttrsToUpdate{Metadata: mdPrepped})
-	if err != nil {
+	if err = w.Close(); err != nil {
 		return nil, err
 	}
 
-	return c.convertToStowItem(attr)
+	return c.convertToStowItem(w.Attrs())
+}
+
+func merge(metadata ...map[string]string) map[string]string {
+	res := map[string]string{}
+	for _, mt := range metadata {
```

Comment:
If the Metadata is key/value pairs according to [google documentation](https://github.com/googleapis/google-cloud-go/blob/storage/v1.23.0/storage/storage.go#L1361). Why do we need the nested loop? Shouldn't we just:
```
for k, v := range metadata {
    res[k] = v
}
```
?

Reply:
This merges a list of maps into one... so the first loop iterates over the list of maps, the second loop iterates over the keys/values in each of them..